### PR TITLE
Changing the defaults to none and validating license picking logic against none

### PR DESCRIPTION
**Pull request by abika5**

### CODE REVIEW

File: mlcluster-vpc.template [hunk 1]
```diff
@@ -357,19 +357,19 @@ Parameters:
...
+    Default: ''
 Conditions:
   UseLogSNS: !Not [!Equals [!Ref LogSNS, "none"]]
   UseSpot: !Not
     - !Equals
       - !Ref SpotPrice
       - 0
   MultiZone: !Not [!Equals [!Ref NumberOfZones, 1]]
-  EssentialEnterprise: !And [!Equals [!Ref LicenseKey, ''], !Equals [!Ref Licensee, '']]
+  EssentialEnterprise: !Not [!And [!Equals [!Ref LicenseKey, ''], !Equals [!Ref Licensee, '']]]
```
Comment (vzhang-progress) [inline mlcluster-vpc.template]: Abika, could you please explain the logic change here?

Thanks,
Vanessa

**Review by vzhang-progress — CHANGES_REQUESTED** [review]
Please see my comments.

Thanks,
Vanessa

**Comment by abika5** [comment]
I think you have commented on the older file. 

The logic behind this line is :
EssentialEnterprise: !Not [!And [!Equals [!Ref LicenseKey, 'none'], !Equals [!Ref Licensee, 'none']]

**! [license key == none && licensee == none ] = EssentialEnterprise**

If  license key = none && licensee =none then EssentialEnterprise = **false** (inturn it picks the developer AMI)
If  license key != none && licensee !=none then EssentialEnterprise = **true** (inturn it picks the Enterprise AMI)

**Comment by yunzvanessa** [comment]
Based on my convo with Sebastian, my understanding is when licenseKey and licensee are both "none", condition EssentialEnterprise is true, which is the original logic. Because EssentialEnterprise means PAYG, the license info is hardcoded in the server. It cannot be changed through CF template, so licenseKey and licensee should both be none. 

I think the original logic: If license key == none && licensee == none then EssentialEnterprise = **true** (pick enterprise AMI) is correct.

Thanks,
Vanessa

**Pull request closed without merging**